sympy/geometry/tests/test_polygon.py

```diff
@@ -522,3 +522,38 @@ def test_second_moment_of_area():
     assert (I_xx - rectangle.second_moment_of_area(p)[0]) == 0
     assert (I_yy - rectangle.second_moment_of_area(p)[1]) == 0
     assert (I_xy - rectangle.second_moment_of_area(p)[2]) == 0
+
+
+def test_cut_section():
+    # concave polygon
+    p = Polygon((-1, -1), (1, S(5)/2), (2, 1), (3, S(5)/2), (4, 2), (5, 3), (-1, 3))
+    l = Line((0, 0), (S(9)/2, 3))
+    p1 = p.cut_section(l)[0]
+    p2 = p.cut_section(l)[1]
+    assert p1 == Polygon(
+        Point2D(-S(9)/13, -S(6)/13), Point2D(1, S(5)/2), Point2D(S(24)/13, S(16)/13),
+        Point2D(S(12)/5, S(8)/5), Point2D(3, S(5)/2), Point2D(S(24)/7, S(16)/7),
+        Point2D(S(9)/2, 3), Point2D(-1, 3), Point2D(-1, -S(2)/3))
+    assert p2 == Polygon(Point2D(-1, -1), Point2D(-S(9)/13, -S(6)/13), Point2D(S(24)/13, S(16)/13),
+        Point2D(2, 1), Point2D(S(12)/5, S(8)/5), Point2D(S(24)/7, S(16)/7), Point2D(4, 2), Point2D(5, 3),
+        Point2D(S(9)/2, 3), Point2D(-1, -S(2)/3))
+
+    # convex polygon
+    p = RegularPolygon(Point2D(0,0), 6, 6)
+    s = p.cut_section(Line((0, 0), slope=1))
+    assert s[0] == Polygon(Point2D(-3*sqrt(3) + 9, -3*sqrt(3) + 9), Point2D(3, 3*sqrt(3)),
+        Point2D(-3, 3*sqrt(3)), Point2D(-6, 0), Point2D(-9 + 3*sqrt(3), -9 + 3*sqrt(3)))
+    assert s[1] == Polygon(Point2D(6, 0), Point2D(-3*sqrt(3) + 9, -3*sqrt(3) + 9),
+        Point2D(-9 + 3*sqrt(3), -9 + 3*sqrt(3)), Point2D(-3, -3*sqrt(3)), Point2D(3, -3*sqrt(3)))
+
+    # case where line does not intersects but coincides with the edge of polygon
+    a, b = 20, 10
+    t1, t2, t3, t4 = [(0, b), (0, 0), (a, 0), (a, b)]
+    p = Polygon(t1, t2, t3, t4)
+    p1, p2 = p.cut_section(Line((0, b), slope=0))
+    assert print(p1) == None
```

Comment:
What is expected here? Apart from the fact that it doesn't work in Python 2.7 such a test doesn't do anything at all.

Did you mean to see the output? Skip the assert and add a Python check if-clause.

Did you mean str? (But that will never return None.)

Reply:
`print` statement mistakenly got there. I will remove it.
I don't know why the local tests didn't fail for this or even the tests for python 3 in travis, never mind I will correct it